[surface] update EOL dates

trogper opened the pull request:
update EOL dates from official website

trogper committed:
Update surface.md

update EOL dates from official website

captn3m0 commented on products/surface.md:
```diff
@@ -12,69 +12,111 @@ releaseColumn: false
-    eol: 2021-11-17
-  - releaseCycle: Surface Pro (5th gen)
-    release: 2017-06-15
+  - releaseCycle: Surface RT1
```
A few rows are incorrectly copied (1 is a footnote marker on the upstream page: https://docs.microsoft.com/en-gb/surface/surface-driver-firmware-lifecycle-support).

captn3m0 replied:
I like that we're reversing the table in source so it matches upstream and makes for easier contributions 👍🏽, but the sorting feels off. Going to try a few options (We want the most supported options to be near the top ideally)

captn3m0 committed:
[surface] Add additional information

- Fixed some footnote mentions
- Added some text explaining the lifecycle
- Attempting a sort by eol

captn3m0 reviewed (approved):
LGTM.

usta commented on products/surface.md:
```diff
@@ -10,71 +10,117 @@ latestColumn: true
 releaseColumn: false
 releaseDateColumn: true
-sortReleasesBy: 'release'
+sortReleasesBy: eol
```
@captn3m0 i know ( at least guessing)  this is working but isnt this tag needs to be a string ? i mean isnt it needs `'  '`?
